pygfx/objects/_base.py

```diff
@@ -133,9 +135,11 @@ def __init__(
         visible=True,
         render_order=0,
         render_mask="auto",
-        position=None,
     ):
         super().__init__()
+        self.up = np.array((0, 1, 0))
```

Comment:
should this be typed as float?

Reply:
Let's make this a property with a setter that does something like `np.asarray(value, dtype=np.float32)`. And then just `self.up = 0, 1, 0` here.

Reply:
It may come as a surprise given my heavy use of it in this PR, but I actually prefer to not use `@property` when I can avoid it :D. They don't combine well with numpy arrays as we lose the ability to assign to slices. We can promote to `float`, but I doubt that type conversion of `up` is a big enough bottleneck to make a measurable difference. 

You call if we want to add, make it a property, or leave it as is.

Reply:
For me, it would be enough to just type it as float :) I don't mind making it a property or not.

Reply:
Solved. While implementing the axis setters I had to push `self.up` into the AffineTransform class (now called `gravity` to avoid aliasing with the `up` axis).

The result is that this is now a `float` ndarray.